pandas/tests/indexes/datetimes/test_constructors.py

```diff
@@ -959,3 +959,78 @@ def test_pass_datetimeindex_to_index(self):
         expected = Index(rng.to_pydatetime(), dtype=object)
 
         tm.assert_numpy_array_equal(idx.values, expected.values)
+
+
+@pytest.mark.parametrize("tz", ["dateutil/Europe/London", "Europe/London"])
+@pytest.mark.parametrize(
+    "ts_input,fold",
+    [
+        (datetime(2019, 10, 27, 1, 30, 0, 0, fold=0), 1),
```

Comment:
I think this is already covered by my other suggestion about the semantics of `Timestamp(..., fold=...)`, but this seems like a problem to me, because it will also hit `datetime(2019, 10, 27, 1, 30), fold=1`.

This creates a somewhat perverse API where when you are passing a naive datetime, a time zone and a `fold` to the `Timestamp` constructor, you'll get an error unless you specify `fold` in *two* places - the `datetime` constructor and the `Timestamp` constructor.

It may be sensible to error on `fold=1` in the `datetime` and `fold=0` in the `Timestamp`, but I think raising exceptions on passing fold to *anything* where `datetime` is already an aware `datetime` will handle all the real cases of ambiguity.

Reply:
Thanks for noticing this. Changed the raise behavior to allow `pd.Timestamp(datetime(2019, 10, 27, 1, 30), fold=1, tz='Europe/London')`.